packages/ui/src/serializers/yaml-camel-resource-serializer.test.ts

```diff
@@ -0,0 +1,32 @@
+import { YamlCamelResourceSerializer } from './yaml-camel-resource-serializer';
+import { camelRouteJson, camelRouteYaml } from '../stubs';
+import { CamelRouteResource } from '../models/camel';
+import { CamelYamlDsl } from '@kaoto/camel-catalog/types';
+
+describe('YamlCamelResourceSerializer', () => {
+  let serializer: YamlCamelResourceSerializer;
+
+  beforeEach(() => {
+    serializer = new YamlCamelResourceSerializer();
+  });
+
+  it('parses YAML code into JSON object', () => {
+    const result = serializer.parse(camelRouteYaml);
+    expect(result).toEqual([camelRouteJson]);
+  });
+
+  it('returns empty array for empty or non-string input in parse', () => {
+    expect(serializer.parse('')).toEqual([]);
+    expect(serializer.parse(null as unknown as string)).toEqual([]);
+    expect(serializer.parse(123 as unknown as string)).toEqual([]);
```

Comment:
Note for the future: check whether returning an empty `[]` is OK for all resources.